### PR TITLE
fix: allow prefix in serve

### PR DESCRIPTION
**Pull request by Caele**
## Motivation

Fix for https://github.com/qlik-oss/nebula.js/issues/1448

## Requirements checklist

- [x] Api specification
  - [x] Ran `yarn spec`
    - [x] No changes **_OR_** API changes has been formally approved
- [ ] Unit/Component test coverage
- [x] Correct PR title for the changes (fix, chore, feat)

When build and tests have passed:

- [ ] Add code reviewers, for example @qlik-oss/nebula-core

### CODE REVIEW

File: commands/serve/web/connect.js [hunk 1]
```diff
@@ -24,7 +24,7 @@ const getParams = () => {
 // QCS:       wss://<tenant-url>.<region>.qlikcloud.com/app/<app-GUID>
 // QSEoK:     wss://<host>/app/<app-GUID>
 // QSEoW:     wss://<host>/<virtual-proxy-prefix>/app/<app-GUID>
-const parseEngineURL = (url, urlRegex = /(wss?):\/\/([^/:?&]+)(?::(\d+))?/, appRegex = /\/app\/([^?&#:]+)/) => {
+const parseEngineURL = (url, urlRegex = /(wss?):\/\/([^/:?&]+)(?::(\d+))?\/?(.*)/, appRegex = /\/app\/([^?&#:]+)/) => {
```
Comment (T-Wizard) [inline commands/serve/web/connect.js:27]: when testing the new urlRegex with example url above.
it looks to me that prefix will also include app. 
Could that not cause problem later in the code?

Comment (Caele) [reply commands/serve/web/connect.js:27]: hmm, yea true, best to strip app first

Comment (Caele) [reply commands/serve/web/connect.js:27]: i think the intended flow works, but the whole thing is not exactly what I thought it was. I'll redo this a bit.

File: commands/serve/web/__tests__/connect.test.js [hunk 2]
```diff
@@ -460,7 +462,38 @@ describe('connect.js', () => {
...
+          host: expect.any(String),
+          port: undefined, // because of providing a link
+          appId: 'SOME_APP_ID/', // since there is an appid in link
+          prefix: 'prefix',
+        }),
+        engineUrl: expect.any(String),
+        appUrl: url,
+      });
+
+      url = `wss://${authConfig.host}/prefix/app/SOME_APP_ID/`;
```
Comment (T-Wizard) [inline commands/serve/web/__tests__/connect.test.js:488]: testing with
url = `wss://${authConfig.host}/app/SOME_APP_ID/`;
give the questionable prefix `app`

Comment (Caele) [reply commands/serve/web/__tests__/connect.test.js:488]: you can likely create a virtual proxy named app, so if you input that it should be fine. Depends on the mode. I don't think the /app part is even used here, as that is a different query param.

Comment (Caele) [reply commands/serve/web/__tests__/connect.test.js:488]: yea I was wrong again, it is indeed used.

**Review by a-m-dev — APPROVED** [review]
LGTM

**Comment by Caele** [comment]
Something else gets really odd, need to rethink this

**Review by T-Wizard — APPROVED** [review]

**Pull request merged**